src/HPolyhedron.jl

```diff
@@ -0,0 +1,206 @@
+
+# constructor for an HPolyhedron with no constraints of type Float64
+HPolyhedron() = HPolyhedron{Float64}()
+
+# constructor for an HPolytope from a simple H-representation
+function HPolyhedron(A::Matrix{N}, b::Vector{N}) where {N<:Real}
+    m = size(A, 1)
+    constraints = LinearConstraint{N}[]
+    @inbounds for i in 1:m
+        push!(constraints, LinearConstraint(A[i, :], b[i]))
+    end
+    return HPolyhedron(constraints)
+end
+
+# convenience union type
+const HPoly{N} = Union{HPolytope{N}, HPolyhedron{N}}
+
+# --- LazySet interface functions ---
+
+"""
+    dim(P::HPoly{N})::Int where {N<:Real}
+
+Return the dimension of a polyhedron in H-representation.
+
+### Input
+
+- `P`  -- polyhedron in H-representation
+
+### Output
+
+The ambient dimension of the polyhedron in H-representation.
+If it has no constraints, the result is ``-1``.
+"""
+function dim(P::HPoly{N})::Int where {N<:Real}
+    return length(P.constraints) == 0 ? -1 : length(P.constraints[1].a)
+end
+
+"""
+    σ(d::AbstractVector{N}, P::HPolyhedron{N}) where {N<:Real}
+
+Return the support vector of a polyhedron (in H-representation) in a given
+direction.
+
+### Input
+
+- `d` -- direction
+- `P` -- polyhedron in H-representation
+
+### Output
+
+The support vector in the given direction.
+
+### Algorithm
+
+This implementation uses `GLPKSolverLP` as linear programming backend.
+"""
+function σ(d::AbstractVector{N}, P::HPolyhedron{N}) where {N<:Real}
+    c = -d
+    n = length(constraints_list(P))
+    @assert n > 0 "the polyhedron has no constraints"
+    A = zeros(N, n, dim(P))
+    b = zeros(N, n)
+    for (i, Pi) in enumerate(constraints_list(P))
+        A[i, :] = Pi.a
+        b[i] = Pi.b
+    end
+    sense = '<'
+    l = -Inf
+    u = Inf
+    solver = GLPKSolverLP()
+    lp = linprog(c, A, sense, b, l, u, solver)
+    if lp.status == :Unbounded
+        error("the support vector in direction $(d) is undefined because " *
+              "the polyhedron is unbounded")
+    elseif lp.status == :Infeasible
+        error("the support vector is undefined because the polyhedron is empty")
+    else
+        return lp.sol
+    end
+end
+
+"""
+    ∈(x::AbstractVector{N}, P::HPoly{N})::Bool where {N<:Real}
+
+Check whether a given point is contained in a polytope in constraint
+representation.
+
+### Input
+
+- `x` -- vector with the coordinates of the point
+- `P` -- polytope in constraint representation
+
+### Output
+
+`true` iff ``x ∈ P``.
+
+### Algorithm
+
+This implementation checks if the point lies on the outside of each hyperplane.
+This is equivalent to checking if the point lies in each half-space.
+"""
+function ∈(x::AbstractVector{N}, P::HPoly{N})::Bool where {N<:Real}
+    @assert length(x) == dim(P)
+
+    for c in P.constraints
+        if dot(c.a, x) > c.b
+            return false
+        end
+    end
+    return true
+end
+
+
+# ===========================================
+# HPolyhedron and HPolytope's shared methods
```

Comment:
Some of the methods above are also shared.

Reply:
thanks, i pulled out all shared methods excepting the conversions

Reply:
But this comment still confuses me.